test/subarray.jl

```diff
@@ -190,7 +190,7 @@ function runtests(A::Array, I...)
     test_bounds(S)
 end
 
-function runtests(A::ANY, I...)
```

Comment:
Is this the method that is causing issue in dispatch? Shouldn't this be less specific than `runtests(name)` in `testdefs.jl` and therefore not be called? Is there a dispatch bug here?

(not that it is a good idea to extend the runtests function anyway so the change lgtm)